src/rviz/ogre_helpers/stl_loader.cpp

```diff
@@ -91,6 +91,13 @@ bool STLLoader::load(const std::string& path)
 bool STLLoader::load(uint8_t* buffer)
 {
   uint8_t* pos = buffer;
+
+  // check for ascii type
+  if(std::string((char*)buffer,5) == std::string("solid")) {
```

Comment:
You don't have to change this, but you should consider avoiding using C-style casting in C++, it's almost always more appropriate to use one of the fixed casting types in C++, in this case:

```c++
if (std::string(reinterpret_cast<char *>(buffer), 5) == std::string("solid")) {
```